File: P1-Linguagem_de_Programação_I/Desafios dos Roteiros - Python/desafio-roteiro06/desafio-roteiro03-final.py
produtos = []

while True:
    print("\n====== SORVETERIA GELADA ======")
    print("1 - Adicionar novo produto")
    print("2 - Listar produtos em estoque")
    print("3 - Sair")

    opcao = input("Escolha uma opção: ")

    if opcao == "1":
        nome = input("Nome do produto: ")
        
        while True:
            quantidade = int(input("Quantidade: "))
            if quantidade >= 0:
                break
            else:
                print("Quantidade não pode ser negativa. Tente novamente.")
        
        preco = float(input("Preço: R$ "))

        produto = {
            "nome": nome,
            "quantidade": quantidade,
            "preco": preco
        }

        produtos.append(produto)

        print("Produto cadastrado com sucesso!")

    elif opcao == "2":
        if len(produtos) == 0:
            print("Nenhum produto cadastrado.")
        else:
            print("\n--- Produtos em Estoque ---")
            for p in produtos:
                print(f"Nome: {p['nome']}")
                print(f"Quantidade: {p['quantidade']}")
                print(f"Preço: R$ {p['preco']:.2f}")
                print("--------------------------")

    elif opcao == "3":
        print("Encerrando o sistema...")
        break

    else:
        print("Opção inválida. Tente novamente.")

File: P1-Linguagem_de_Programação_I/Desafios dos Roteiros - C++/desafio-roteiro03-final.cpp
#include <iostream>
#include <vector>
using namespace std;

struct Produto {
    string nome;
    int quantidade;
    float preco;
};

int main() {

    vector<Produto> produtos;
    int opcao;

    while (true) {

        cout << "\n====== SORVETERIA GELADA ======\n";
        cout << "1 - Adicionar novo produto\n";
        cout << "2 - Listar produtos em estoque\n";
        cout << "3 - Sair\n";
        cout << "Escolha uma opcao: ";
        cin >> opcao;

        if (opcao == 1) {

            Produto p;

            cout << "Nome do produto: ";
            cin >> p.nome;

            do {
                cout << "Quantidade: ";
                cin >> p.quantidade;

                if (p.quantidade < 0) {
                    cout << "Quantidade nao pode ser negativa. Tente novamente.\n";
                }

            } while (p.quantidade < 0);

            cout << "Preco: R$ ";
            cin >> p.preco;

            produtos.push_back(p);

            cout << "Produto cadastrado com sucesso!\n";
        }

        else if (opcao == 2) {

            if (produtos.size() == 0) {
                cout << "Nenhum produto cadastrado.\n";
            }
            else {

                cout << "\n--- Produtos em Estoque ---\n";

                for (int i = 0; i < produtos.size(); i++) {

                    cout << "Nome: " << produtos[i].nome << endl;
                    cout << "Quantidade: " << produtos[i].quantidade << endl;
                    cout << "Preco: R$ " << produtos[i].preco << endl;
                    cout << "--------------------------\n";
                }
            }
        }

        else if (opcao == 3) {
            cout << "Encerrando o sistema...\n";
            break;
        }

        else {
            cout << "Opcao invalida. Tente novamente.\n";
        }
    }

    return 0;
}